models/repo.go

```diff
@@ -1524,11 +1556,15 @@ func updateRepository(e Engine, repo *Repository, visibilityChanged bool) (err e
 
 		// Create/Remove git-daemon-export-ok for git-daemon...
 		daemonExportFile := path.Join(repo.RepoPath(), `git-daemon-export-ok`)
-		if repo.IsPrivate && com.IsExist(daemonExportFile) {
+		isExist, err := util.IsExist(daemonExportFile)
+		if err != nil {
+			log.Error("Unable to check if %s exists. Error: %v", daemonExportFile, err)
```

Comment:
```return err```, in next code you check ```!isExist``` bu not check ```err```

Reply:
Hmm... The issue is not the recreation of the export file here - it's the non-deletion of the file is the repo has gone private. So on reflection I will return the error here.

```suggestion
			log.Error("Unable to check if %s exists. Error: %v", daemonExportFile, err)
			return err
```